app/controllers/login_controller.rb

```diff
@@ -4,92 +4,44 @@
 require 'date'
 
 class LoginController < ApplicationController
-  # def callback
-  #   # Grab the value of the authorization code returned by Procore
-  #   @authorization_code = params["code"]
-
-  #   # Populate the request body with the defined parameters from the docs
-  #   # Reference documentation: https://developers.procore.com/reference/authentication
-  #   request = {
-  #     grant_type: "authorization_code",
-  #     client_id: ENV['CLIENT_ID'],
-  #     client_secret: ENV['CLIENT_SECRET'],
-  #     code: @authorization_code,
-  #     redirect_uri: ENV['REDIRECT_URI']
-  #   }
-
-  #   # Send a request for an access token to Procore and store the response in a session variable
-  #   response = RestClient.post(ENV['OAUTH_URL'] +'/oauth/token', request.to_json, { content_type: :json, accept: :json })
-  #   session[:oauth_response] = JSON.parse(response)
-
-  #   # Redirect the user's browser to the intended home page
-  #   redirect_to users_home_path, success: "You have successfully obtained an access token!"
-
-  #   rescue RestClient::ExceptionWithResponse
-  #     if session[:oauth_response]
-  #       redirect_to users_home_path, danger: "Something went wrong. Please refresh your access token and try again."
-  #     else
-  #       redirect_to login_index_path, danger: "Something went wrong. Please try again."
-  #     end
-  #   end
+  rescue_from Procore::Error do |exception|
```

Comment:
I saw in the previous commit that `Procore::Error` was _specifically_ rescued by several controlleres.

What is the case for this "generic" `rescue_from`?
Which actions _might_ raise a `Procore::Error` but don't rescue it?

Reply:
It's a generic error from the RubySDK. Anything that's making an API request with the SDK could raise it, and those should all be rescued.